pandas/core/nanops.py

```diff
@@ -718,6 +718,9 @@ def reduction(values, axis=None, skipna=True, mask=None):
                 result = np.nan
         else:
             result = getattr(values, meth)(axis)
+            if (is_integer(result) and is_datetime_or_timedelta_dtype(dtype)
```

Comment:
this needs handling not here but in _wrap_resulf where a scalar should be turned into NaT if it’s null and of the correct dtype